zeppelin-zengine/src/main/java/org/apache/zeppelin/storage/FileSystemConfigStorage.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.apache.zeppelin.storage;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.apache.hadoop.fs.Path;
+import org.apache.zeppelin.conf.ZeppelinConfiguration;
+import org.apache.zeppelin.helium.HeliumConf;
+import org.apache.zeppelin.interpreter.InterpreterInfoSaving;
+import org.apache.zeppelin.interpreter.InterpreterSetting;
+import org.apache.zeppelin.notebook.FileSystemStorage;
+import org.apache.zeppelin.notebook.NotebookAuthorizationInfoSaving;
+import org.apache.zeppelin.user.CredentialsInfoSaving;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * It could be used either local file system or hadoop distributed file system,
+ * because FileSystem support both local file system and hdfs.
+ *
+ */
+public class FileSystemConfigStorage extends ConfigStorage {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FileSystemConfigStorage.class);
+
+  private FileSystemStorage fs;
+  private Path interpreterSettingPath;
+  private Path authorizationPath;
+  private Path credentialPath;
+
+  public FileSystemConfigStorage(ZeppelinConfiguration zConf) throws IOException {
+    super(zConf);
+    this.fs = FileSystemStorage.get(zConf);
+    this.fs.tryMkDir(new Path(zConf.getConfigFSDir()));
```

Comment:
This is incorrect. `FileSystemStorage.get(zconf)` gives you an FS for zconf.getNotebookDir(). The notebook directory is *not* guaranteed to be on the same FS. Context: I tried to use the local `conf/` directory for zeppelin configuration, but put the notebook directory on a `gs://` path so that I could persistently save the notebook.

The same applies to interpreterSettingPath, authorizationPath, credentialPath.

FWIW, I think it's a mistake for FileSystemStorage/FileSystemNotebookRepo to use zconf.getNotebookDir(). Instead, there should be a separate config like zconf.getHcfsNotebookDir(). That way, each storage implementation has its own independent configuration. Right now, you cannot use FileSystemNotebookRepo with GitNotebookRepo, because they rely on the same configuration.

I'm happy to file JIRAs for these if you would like.